tools/schema_loader.cc

```diff
@@ -0,0 +1,128 @@
+#include "locator/token_metadata.hh"
+#include "locator/snitch_base.hh"
+#include "tools/schema_loader.hh"
+#include "utils/fb_utilities.hh"
+
+namespace {
+
+sstring read_file(std::filesystem::path path) {
+    auto file = open_file_dma(path.native(), open_flags::ro).get();
+    auto fstream = make_file_input_stream(file);
+
+    size_t size = 0;
+    std::vector<temporary_buffer<char>> bufs;
+    while (auto buf = fstream.read().get()) {
+        size += buf.size();
+        bufs.emplace_back(std::move(buf));
+    }
+
+    sstring content(sstring::initialized_later{}, size);
+    char* it = content.begin();
+    for (const auto& buf : bufs) {
+        it = std::copy_n(buf.begin(), buf.size(), it);
+    }
+    return content;
+}
+
+std::vector<sstring> split_statements(sstring statements) {
+    std::vector<sstring> split_res;
+    boost::split(split_res, statements, [] (char c) { return c == ';'; });
+    std::vector<sstring> res;
+    for (auto& stmt : split_res) {
+        if (!std::all_of(stmt.begin(), stmt.end(), [] (const char& c) { return std::isspace(c); })) {
+            res.emplace_back(std::move(stmt));
+        }
+    }
+    return res;
+}
+
+schema_ptr do_load_schema(std::filesystem::path path) {
+    cql3::cql_stats cql_stats;
+
+    db::config cfg;
+    cfg.enable_cache(false);
+    cfg.volatile_system_keyspace_for_testing(true);
+
+    database_config dbcfg;
+    dbcfg.available_memory = 1'000'000'000; // 1G
+    service::migration_notifier migration_notifier;
+    gms::feature_service feature_service(gms::feature_config_from_db_config(cfg));
+    locator::shared_token_metadata token_metadata;
+    abort_source as;
+    sharded<semaphore> sst_dir_sem;
+
+    sst_dir_sem.start(1).get();
+    auto sst_dir_sem_stop = deferred_stop(sst_dir_sem);
+
+    utils::fb_utilities::set_broadcast_address(gms::inet_address(0x7f000001)); // 127.0.0.1
+    locator::i_endpoint_snitch::create_snitch(cfg.endpoint_snitch()).get();
+
+    database db(cfg, dbcfg, migration_notifier, feature_service, token_metadata, as, sst_dir_sem);
+    auto stop_db = deferred_stop(db);
+
+    schema_ptr schema;
+
+    for (const auto& statement_str : split_statements(read_file(path))) {
+        auto raw_statement = cql3::query_processor::parse_statement(statement_str);
+        auto prepared_statement = raw_statement->prepare(db, cql_stats);
+        auto* statement = prepared_statement->statement.get();
+
+        if (auto p = dynamic_cast<cql3::statements::create_keyspace_statement*>(statement)) {
+            db.create_keyspace(p->get_keyspace_metadata(*token_metadata.get())).get();
+        } else if (auto p = dynamic_cast<cql3::statements::create_type_statement*>(statement)) {
+            auto type = p->create_type(db);
+            db.find_keyspace(p->keyspace()).add_user_type(std::move(type));
+        } else if (auto p = dynamic_cast<cql3::statements::create_table_statement*>(statement)) {
+            schema = p->get_cf_meta_data(db);
```

Comment:
Probably better to return a vector of schemas, a .cql file can contain multiple tables.